pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -169,9 +171,40 @@ func (c *CSIMaxVolumeLimitChecker) getCSIDriver(pvc *v1.PersistentVolumeClaim) (
 
 	csiSource := pv.Spec.PersistentVolumeSource.CSI
 	if csiSource == nil {
-		klog.V(5).Infof("Not considering non-CSI volume %s/%s", namespace, pvcName)
-		return placeHolderCSIDriver, placeHolderHandle
+		if !csilib.IsPVMigratable(pv) {
+			klog.V(5).Infof("Non-CSI volume %s/%s is not migratable", namespace, pvcName)
+			return "", ""
+		}
+
+		pluginName, err := csilib.GetInTreePluginNameFromSpec(pv, nil)
+		if err != nil {
+			klog.V(5).Infof("Unable to look up plugin name from PV spec: %v", err)
+			return "", ""
+		}
+
+		if !isCSIMigrationOn(csiNode, pluginName) {
+			klog.V(5).Infof("CSI Migration of plugin %s is not enabled", pluginName)
+			return "", ""
+		}
+
+		csiPV, err := csilib.TranslateInTreePVToCSI(pv)
```

Comment:
Are we also handling inline (non-pv) migration?

Reply:
That's fine. Migration is still alpha this release. I think most important is to convert existing CSI limits to use the new csinode field.